diff --git a/src/meson.build b/src/meson.build
--- a/src/meson.build
+++ b/src/meson.build
@@ -303,7 +303,7 @@ if get_option('gtk3')
     libvte_gtk3_sources += a11y_sources
   endif
 
-  libvte_gtk3 = shared_library(
+  libvte_gtk3 = library(
     vte_gtk3_api_name,
     sources: libvte_gtk3_sources,
     version: libvte_gtk3_soversion,
@@ -349,7 +349,7 @@ if get_option('gtk4')
   libvte_gtk4_deps = libvte_common_deps + [gtk4_dep,]
   libvte_gtk4_public_deps = libvte_common_public_deps + [gtk4_dep,]
 
-  libvte_gtk4 = shared_library(
+  libvte_gtk4 = library(
     vte_gtk4_api_name,
     sources: libvte_gtk4_sources,
     version: vte_gtk4_soversion,
