diff --git a/meson_options.txt b/meson_options.txt
index b7b6b382..fa9e82d8 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -22,6 +22,13 @@ option(
   description: 'Use -Bsymbolic-functions',
 )
 
+option(
+  'build-app',
+  type: 'boolean',
+  value: true,
+  description: 'Build the VTE demo application',
+)
+
 option(
   'a11y',
   type: 'boolean',
diff --git a/src/meson.build b/src/meson.build
index 3f89f492..bc45afbf 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -303,7 +303,7 @@ if get_option('gtk3')
     libvte_gtk3_sources += a11y_sources
   endif
 
-  libvte_gtk3 = shared_library(
+  libvte_gtk3 = library(
     vte_gtk3_api_name,
     sources: libvte_gtk3_sources,
     version: libvte_gtk3_soversion,
@@ -349,7 +349,7 @@ if get_option('gtk4')
   libvte_gtk4_deps = libvte_common_deps + [gtk4_dep,]
   libvte_gtk4_public_deps = libvte_common_public_deps + [gtk4_dep,]
 
-  libvte_gtk4 = shared_library(
+  libvte_gtk4 = library(
     vte_gtk4_api_name,
     sources: libvte_gtk4_sources,
     version: libvte_gtk4_soversion,
@@ -746,4 +746,6 @@ endif
 
 # Demo application
 
-subdir('app')
+if get_option('build-app')
+  subdir('app')
+endif
