diff --git a/src/main.cpp b/src/main.cpp
index 8727109ee..ebd3f49bd 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -143,10 +143,2 @@ int main(int argc, char *argv[])
 #endif
-
-    /**
-     * trigger initialisation of proper icon theme
-     */
-#if KICONTHEMES_VERSION >= QT_VERSION_CHECK(6, 3, 0)
-    KIconTheme::initTheme();
-#endif
-
 #if HAVE_DBUS
@@ -160,5 +152,9 @@ int main(int argc, char *argv[])
 #endif
-
     auto app = new QApplication(argc, argv);
-
+    /**
+     * trigger initialisation of proper icon theme
+     */
+#if KICONTHEMES_VERSION >= QT_VERSION_CHECK(6, 3, 0)
+    KIconTheme::initTheme();
+#endif
 #if HAVE_STYLE_MANAGER
